### PR TITLE
feat: fix avatar caching issue 

### LINKED ISSUE #746: Bug: getAvatarLink is using the redirect link
### Describe the bug

Proposal would be diverging the existing `getAvatarLink` into `getAvatarByUsername` and another meta function `getAvatarById`, then slowly offsetting `getAvatarByUsername` in favour of the more efficient `getAvatarById`

### Steps to reproduce

`getAvatarLink` is using the aliased `username` link that redirects to the `user_id` - this is a neat way to figure out a user's github id by looking at their picture, but not a good way to load pictures.

### Affected services

insights.opensauced.pizza

### Platforms

_No response_

### Browsers

_No response_

### Environment

_No response_

### Additional context

tooling, QoL, !GFI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### LINKED ISSUE #757: Feature: Update avatar caching
### Type of feature

🍕 Feature

### Current behavior

Currently, avatars shown in the chart are cached in Cloudinary, while the ones in the hover card are cached in the API. 
This can result in mismatched avatars being displayed on different pages of the site.

| avatar_1_edit | avatar_2_edit |
| --------- | --------- |

### Suggested solution

Maybe having a sort of consistent caching strategy for avatars across all pages of the site may solve this issue :thinking: 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by TAKANOME-DEV**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR:
- replaces the github avatar URL `https://www.github.com/${username}.png?size=60` with the cached avatar `roundedImage(https://www.github.com/${username}.png?size=60, process.env.NEXT_PUBLIC_CLOUD_NAME)`
- renames `getAvatarLink` to `getAvatarByUsername`
- adds `getAvatarById` that uses the correct github avatar link (see #746 )
## Related Tickets & Documents

Fixes #757 
Fixes #746 
## Mobile & Desktop Screenshots/Recordings

| Before | After |
| --------- | --------- |
| before | after |

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: .dccache [hunk 1]
```diff
@@ -0,0 +1 @@
..."],"/home/takanome_dev/Projects/OpenSource/open-sauced/insights/lib/hooks/useFetchUser.ts":[471,1674743190271.6147,"07868fa2a1109834c9d4f084fbb1219a9d4134a4692e44ca3f6b38b5558a6762"],"/home/takanome_dev/Projects/OpenSource/open-sauced/insights/stories/atoms/insight-badge.stories.tsx":[459,1674743190283.6147,"0d40ea91a0041ba44f5c69802eabb4997d58f6e1ba5a8953c51421b941a42da2"],"/home/takanome_dev/Projects/OpenSource/open-sauced/insights/stories/atoms/language-pill.stories.tsx":[614,1674743190283.6147,"b00134689fde7040722873cf18317897cfb1ab9d87d422d2b049541885e5cc0c"],"/home/takanome_dev/Projects/OpenSource/open-sauced/insights/stories/molecules/pull-request-social-card.stories.tsx":[1087,1674743190283.6147,"9ff3212e55a47cde3dd8ae90c81f2bd60e13dce5053f9366ad2bed4a59761266"],"/home/takanome_dev/Projects/OpenSource/open-sauced/insights/stories/organisms/user-settings-page.stories.tsx":[408,1674743190283.6147,"044de26aac0f81afea419335c92df49d9d35d3ced082c5beb7dec24368857745"],"/home/takanome_dev/Projects/OpenSource/open-sauced/insights/components/organisms/UserSettingsPage/user-settings-page.tsx":[4914,1674743190263.6145,"354d2178990c897487a58f2d5212c0312623037d27c6533ebe04aa3285438505"]}
```
Comment (brandonroberts) [inline .dccache]: Delete this?

File: .gitignore [hunk 2]
```diff
@@ -277,3 +277,4 @@ storybook-static
 **/__snapshots__
 .env*
 !.env
+.dccache
```
Comment (brandonroberts) [inline .gitignore:280]: This seems unrelated to these changes. Why is this here?

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: .dccache (hunk 1 above)
Comment (TAKANOME-DEV) [reply .dccache]: Yeah, I thought I removed it. Thanks :+1:

**Review by OgDev-01 — APPROVED** [review]
LGTM 🍕

File: components/atoms/Avatar/avatar-hover-card.tsx [hunk 3]
```diff
@@ -4,14 +4,16 @@ import Link from "next/link";
...
+import { getAvatarByUsername } from "lib/utils/github";
+import roundedImage from "lib/utils/roundedImages";
 
 export declare interface AvatarProps {
   contributor: string;
   repositories: number[];
 }
 
 const AvatarHoverCard = ({ contributor, repositories }: AvatarProps): JSX.Element => {
+  const avatar = roundedImage(getAvatarByUsername(contributor), process.env.NEXT_PUBLIC_CLOUD_NAME);
```
Comment (brandonroberts) [inline components/atoms/Avatar/avatar-hover-card.tsx:16]: Maybe a separate issue, but we should default the `cloudName` argument to the `process.env.NEXT_PUBLIC_CLOUD_NAME` so we don't have to pass it around everywhere.

```suggestion
  const avatar = roundedImage(getAvatarByUsername(contributor));
```

File: components/molecules/HoverCardWrapper/hover-card-wrapper.tsx [hunk 4]
```diff
@@ -25,7 +27,7 @@ const HoverCardWrapper = ({ username, repositories }: HoverCardWrapperProps) =>
   });
 
   const profile: ContributorsProfileType = {
-    githubAvatar: `https://www.github.com/${username}.png?size=60`,
+    githubAvatar: roundedImage(`https://www.github.com/${username}.png?size=60`, process.env.NEXT_PUBLIC_CLOUD_NAME),
```
Comment (brandonroberts) [inline components/molecules/HoverCardWrapper/hover-card-wrapper.tsx:30]: Why not use the `getAvatarById` function here?

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: components/molecules/HoverCardWrapper/hover-card-wrapper.tsx (hunk 4 above)
Comment (TAKANOME-DEV) [reply components/molecules/HoverCardWrapper/hover-card-wrapper.tsx:30]: The id of the user is not available there. Most of the places where this component is used don't have the id of the user, only the name

File: components/atoms/Avatar/avatar-hover-card.tsx (hunk 3 above)
Comment (TAKANOME-DEV) [reply components/atoms/Avatar/avatar-hover-card.tsx:16]: Yeah, totally. Let me update it :rocket:

File: components/molecules/HoverCardWrapper/hover-card-wrapper.tsx (hunk 4 above)
Comment (brandonroberts) [reply components/molecules/HoverCardWrapper/hover-card-wrapper.tsx:30]: Ahh, I see
